src/Http/Http.Features/src/IHeaderDictionary.Keyed.cs

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+#if NET6_0_OR_GREATER
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Http
+{
+    public partial interface IHeaderDictionary
```

Comment:
Do we really want this list to be exhaustive? We should have specific criteria documented at the top of the file for what should and shouldn't be added here. The same goes for HeaderNames.

Reply:
>  A big downside to this approach is there's no distinction between request and response headers. 

No different than now? Could change the features to return two different IHeader dictionaries if preferred; but it would make utility methods more complicated.

> both request and response headers types will have Cookie and SetCookie properties and that will confuse people about which is the correct header to use.

Bad example? The people that will get confused by that will likely use `Response.Cookies` and `Request.Cookies`; and even though I know the difference between the two I use the features rather than actually using the headers as aint no bdy got time for manual parsing splitting and reforming the two headers?

Reply:
It's slightly worse than today as today you could pass anything into the indexer but we don't explicitly recommend anything. Adding these APIs makes them more visible and more likely to be used incorrectly.

I agree that the solutions like splitting the type are worse than the problem. More basic mitigations like adding doc comments for each header might be adequate for now.